lib/src/lifecycle_module.dart

```diff
@@ -272,6 +287,12 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
   /// [onLoad] call will be emitted.
   Stream<LifecycleModule> get didLoad => _didLoadController.stream;
 
+  /// The [LifecycleModule] has been declared firstUseful
+  ///
+  /// Used for the parent module to know that its child is useful, as oftentimes
+  /// a parent can't be called "Useful" until all of its children are
+  Stream<LifecycleModule> get isFirstUseful => _isFirstUsefulController.stream;
```

Comment:
Do you have an example of how this will be used in practice?

Reply:
Here is our use case: https://github.com/Workiva/graph_app/pull/7145/files#diff-67c7c5d0ea4739dbee0a0091e12a0dbdR83

The idea is that in practice we don't have a perfect modular hierarchy. Graph_app parent modules might have children that are modules and children that are strictly dumb components, but we don't want to specify the overarching parent module as useful until both the child modules and dumb components are "first useful", so we track both manual spans and child.isFirstUseful